src/cryptography/x509/extensions.py

```diff
@@ -134,38 +137,60 @@ def get_extension_for_class(self, extclass) -> "Extension[ExtensionType]":
             if isinstance(ext.value, extclass):
                 return ext
 
+        oid = typing.cast(ObjectIdentifier, extclass.oid)
         raise ExtensionNotFound(
-            "No {} extension was found".format(extclass), extclass.oid
+            "No {} extension was found".format(extclass), oid
         )
 
-    __len__, __iter__, __getitem__ = _make_sequence_methods("_extensions")
+    @typing.overload
```

Comment:
Let's hold off on making this change. Everything else in this PR is unobjectionable and I'm happy to merge, but changing all of our extension types to be implemented like this is something I think we should hold off on.

Reply:
Not sure what _exactly_ you want reverted here? "manually" implementing the sequence methods? Just the overload?

Reply:
Stick with `_make_sequence_methods` and then a seperate PR to discuss these changes.

Reply:
done!